### DISCUSSION TITLE
Enhancement: Local/Online multiplayer?

### DISCUSSION THREAD

**Discussion by BojayB** [FunkinCrew/Funkin#1589] [Ideas]
As the title suggests, I think this game would be crazy fun to play with someone else! Either:
* You or multi-buddy controls either baddie of whichever song or BF, maybe there could be a character select (similar to controller select in multiplayer Guitar Hero 2 (img below)) at the beginning of each level or a preset you could select as your default character of choice when in said level.

    OR

* You or multi-buddy controls BF or {new character implemented purely for multi??} while you both face off against whichever baddie. Only problem with that would be creating more sprites, and a second voice to overlay the bf while he sings so both can play at the same time / duet like bf and baddie. Character select still applies?

I would love to use this multiplayer mode to be able to host FNF tournaments in discord servers with friends. I think it would be really fun!

![image](https://user-images.githubusercontent.com/41974762/107445068-be81fe00-6b09-11eb-94c3-7c054b138bc3.png)
^ guitar hero layout

**Comment by oketariq** [comment]
My first thought when i saw this was "Just, no."

Local multiplayer is a mod, and Online multiplayer would be hard for Ninjamuffin. He already barely earns anything off of the game since it's out for free and the only cash he earns from it are donations and it would just be a hassle to setup servers. Even if it was done, the servers would most likely not even be strong enough to hold the massive amount of people going onto it at the same time.

**Comment by c3ypt1c** [comment]
@hdboye

You're totally right. Implementing anything more than local multiplayer might be a huge hassle for little reward, and you kind of cannot use servers for the simple fact of additional latency. Your opponent would likely be delayed by a couple ms depending on your ping relative to each other and I'm not sure how noticeable that would be when playing songs.

**Comment by BojayB** [comment]
Then what about local multiplayer that could be played via personal servers that the user has setup or something like hamachi? Or just standard local multiplayer.

**Comment by BakedCrossiant** [comment]
https://peerjs.com/

**Comment by BojayB** [comment]
> My first thought when i saw this was "Just, no."
> 
> 
> 
> Local multiplayer is a mod, and Online multiplayer would be hard for Ninjamuffin. He already barely earns anything off of the game since it's out for free and the only cash he earns from it are donations and it would just be a hassle to setup servers. Even if it was done, the servers would most likely not even be strong enough to hold the massive amount of people going onto it at the same time.

Bet

**Comment by ghost** [comment]
There is an FNF multiplayer mod
And there's a localhost redirect thing. Start the server through fnf multi to localhost and then others can connect

**Comment by BojayB** [comment]
My point was that the full game will include multiplayer
